unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NavMap/NavMapLocationControlsController.cs

```diff
@@ -0,0 +1,93 @@
+﻿using DCL.Helpers;
+using DCLServices.MapRendererV2.ConsumerUtils;
+using DCLServices.MapRendererV2.MapCameraController;
+using System;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace DCL
+{
+    public class NavMapLocationControlsController : IDisposable
+    {
+        private const float TRANSLATION_DURATION = 0.5f;
+        private readonly NavMapLocationControlsView view;
+        private readonly NavmapZoomViewController navmapZoomViewController;
+        private readonly NavmapToastViewController toastViewController;
+
+        private readonly MapRenderImage.ParcelClickData homeParcel = new ()
+        {
+            Parcel = DataStore.i.HUDs.homePoint.Get(),
+            WorldPosition = new Vector2(Screen.width / 2f, Screen.height / 2f),
+        };
+        private IMapCameraController mapCamera;
+
+        private bool active;
+        private CancellationTokenSource cts;
+
+        public NavMapLocationControlsController(NavMapLocationControlsView view, NavmapZoomViewController navmapZoomViewController, NavmapToastViewController toastViewController)
+        {
+            this.view = view;
+            this.navmapZoomViewController = navmapZoomViewController;
+            this.toastViewController = toastViewController;
+        }
+
+        public void Dispose()
+        {
+            cts?.Cancel();
+            cts?.Dispose();
+        }
+
+        public void Activate(IMapCameraController mapCameraController)
+        {
+            if (active && mapCamera == mapCameraController)
+                return;
+
+            cts = new CancellationTokenSource();
+
+            mapCamera = mapCameraController;
+
+            view.homeButton.onClick.AddListener(FocusOnHomeLocation);
+            view.centerToPlayerButton.onClick.AddListener(CenterToPlayerLocation);
+
+            active = true;
+        }
+
+        public void Deactivate()
+        {
+            if (!active) return;
+
+            cts.Cancel();
+            cts.Dispose();
+            cts = null;
```

Comment:
I see we are adopting the cts.SafeRestart(); function in more places, if you want you can use it as well